### PR TITLE
Removed AutoSubscribePlainMessages as settings option

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Updates related to https://github.com/Particular/NServiceBus/pull/3284

### CODE REVIEW

**Comment by danielmarbach** [comment]
I think we should also hint in the upgrade guide

**Comment by dvdstelt** [comment]
Added additional section to upgrade guide

**Comment by dvdstelt** [comment]
Ah, apparently only 1 can be assigned :smile:

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md [hunk 1]
```diff
@@ -37,10 +35,11 @@ snippet:DoNotAutoSubscribeSagas
 
 ### Auto subscribe to plain messages
 
-Before Version 4 all messages not defined as a command using `ICommand` or the `.DefiningCommandsAs` convention where automatically subscribed. You can opt-in to the old behavior using:
+Before Version 4 all messages not defined as a command using `ICommand` or the `.DefiningCommandsAs` convention where automatically subscribed. In versions 4 and 5 you could opt-into that legacy behavior using following code
 
 snippet:AutoSubscribePlainMessages
 
+Starting with version 6 we removed that possibility as it was encouraging bad practices.
```
Comment (SimonCropp) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: remove the "we"

Comment (SimonCropp) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: > as it was encouraging bad practices

you cant just drop this hear with no more detail

File: nservicebus/upgrades/5to6.md [hunk 2]
```diff
@@ -167,6 +167,9 @@ The `IMessageMutator` and `IMutateTransportMessages` interfaces are no longer av
 
 `ISubscriptionStorage` implements the concern of storage, retrieval and removal for subscriptions, which is executed inside the context of a pipeline. Furthermore, `ISubscriptionStorage` introduced a new parameter `SubscriptionStorageOptions`. This parameter allows access to the pipeline context. This enables subscription storages to manipulate everything that exists in the context during message pipeline execution.
 
+### Auto subscribing plain messages
+
+The option to automatically subscribe to plain messages was removed to discourage bad practices. Message subscription should only be based on events. Although not recommended, this can be overridden by [manually subscribing](http://docs.particular.net/nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed#how-to-manually-subscribe-to-a-message) to other message types.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: > http://docs.particular.net/nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed#how-to-manually-subscribe-to-a-message

dont fully qualify links https://github.com/Particular/docs.particular.net#linking

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md [hunk 3]
```diff
@@ -37,10 +35,11 @@ snippet:DoNotAutoSubscribeSagas
 
 ### Auto subscribe to plain messages
 
-Before Version 4 all messages not defined as a command using `ICommand` or the `.DefiningCommandsAs` convention where automatically subscribed. You can opt-in to the old behavior using:
+Before Version 4 all messages not defined as a command using `ICommand` or the `.DefiningCommandsAs` convention where automatically subscribed. In versions 4 and 5 you could opt-into that legacy behavior using following code
```
Comment (SimonCropp) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: > Before Version 4

this is ambiguous. ie is it inclusive of v4 or not.

> In Version 4 and lower...

File: nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md (hunk 1 above)
Comment (dvdstelt) [inline nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md]: Just removed it, but will change it with more detail. :+1:

**Pull request merged**